### ISSUE TITLE
SQL interpreter catalog completion failure

### ISSUE THREAD

**Issue by indig** [polynote/polynote#99]
SQL interpreter failed to complete using the Spark catalog in the demo today. I was able to reproduce by running a cell with
```scala
spark.catalog.listDatabases().map(_.name).collect()
```

Error: 
```
Job aborted due to stage failure: Task 6 in stage 9.0 failed 4 times, most recent failure: Lost task 6.3 in stage 9.0 (TID 97, 100.85.124.45, executor 3): java.lang.ClassCastException: cannot assign instance of scala.collection.immutable.List$SerializationProxy to field org.apache.spark.rdd.RDD.org$apache$spark$rdd$RDD$$dependencies_ of type scala.collection.Seq in instance of org.apache.spark.rdd.MapPartitionsRDD at java.io.ObjectStreamClass$FieldReflector.setObjFieldValues(ObjectStreamClass.java:2287) at java.io.ObjectStreamClass.setObjFieldValues(ObjectStreamClass.java:1417) at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2293) at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2211) at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2069) at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1573) at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2287) at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2211) at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2069) at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1573) at java.io.ObjectInputStream.readObject(ObjectInputStream.java:431) at scala.collection.immutable.List$SerializationProxy.readObject(List.scala:479) at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1170) at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2178) at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2069) at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1573) at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2287) at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2211) at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2069) at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1573) at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2287) at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2211) at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2069) at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1573) at java.io.ObjectInputStream.readObject(ObjectInputStream.java:431) at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:75) at org.apache.spark.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:114) at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:80) at org.apache.spark.scheduler.Task.run(Task.scala:100) at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:353) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:748) Driver stacktrace: (org.apache.spark.SparkException)
org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1455)
org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1443)
org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1442)
scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1442)
org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:811)
org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:811)
scala.Option.foreach(Option.scala:257)
org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:811)
org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1670)
org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1625)
org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1614)
org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:628)
org.apache.spark.SparkContext.runJob(SparkContext.scala:1930)
org.apache.spark.SparkContext.runJob(SparkContext.scala:1943)
org.apache.spark.SparkContext.runJob(SparkContext.scala:1956)
org.apache.spark.SparkContext.runJob(SparkContext.scala:1970)
org.apache.spark.rdd.RDD$$anonfun$collect$1.apply(RDD.scala:941)
org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
org.apache.spark.rdd.RDD.withScope(RDD.scala:367)
org.apache.spark.rdd.RDD.collect(RDD.scala:940)
org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:275)
org.apache.spark.sql.Dataset$$anonfun$org$apache$spark$sql$Dataset$$execute$1$1.apply(Dataset.scala:2390)
org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:75)
org.apache.spark.sql.Dataset.withNewExecutionId(Dataset.scala:2799)
org.apache.spark.sql.Dataset.org$apache$spark$sql$Dataset$$execute$1(Dataset.scala:2389)
org.apache.spark.sql.Dataset$$anonfun$org$apache$spark$sql$Dataset$$collect$1.apply(Dataset.scala:2394)
org.apache.spark.sql.Dataset$$anonfun$org$apache$spark$sql$Dataset$$collect$1.apply(Dataset.scala:2394)
org.apache.spark.sql.Dataset.withCallback(Dataset.scala:2812)
org.apache.spark.sql.Dataset.org$apache$spark$sql$Dataset$$collect(Dataset.scala:2394)
org.apache.spark.sql.Dataset.collect(Dataset.scala:2370)
$notebook1.Eval$Cell1$3$.<init>(Cell1:2)
$notebook1.Eval$Cell1$3$.<clinit>(Cell1:-1)
java.lang.Class.forName0(Class.java:-2)
java.lang.Class.forName(Class.java:348)
scala.reflect.runtime.ReflectionUtils$.staticSingletonInstance(ReflectionUtils.scala:64)
scala.reflect.runtime.JavaMirrors$JavaMirror$JavaModuleMirror.instance(JavaMirrors.scala:528)
polynote.kernel.lang.scal.ScalaInterpreter$$anonfun$runCode$1$$anonfun$apply$7$$anonfun$apply$8$$anonfun$4.apply(ScalaInterpreter.scala:92)
polynote.kernel.lang.scal.ScalaInterpreter$$anonfun$runCode$1$$anonfun$apply$7$$anonfun$apply$8$$anonfun$4.apply(ScalaInterpreter.scala:84)
cats.effect.IO$Map.apply(IO.scala:1412)
cats.effect.IO$Map.apply(IO.scala:1408)
cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:139)
cats.effect.internals.IORunLoop$.startCancelable(IORunLoop.scala:41)
cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:86)
cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:70)
cats.effect.internals.Trampoline.startLoop(Trampoline.scala:36)
cats.effect.internals.TrampolineEC$JVMTrampoline.cats$effect$internals$TrampolineEC$JVMTrampoline$$super$startLoop(TrampolineEC.scala:93)
cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply$mcV$sp(TrampolineEC.scala:93)
cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply(TrampolineEC.scala:93)
cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply(TrampolineEC.scala:93)
scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:92)
cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:44)
cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:72)
cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:52)
cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:136)
cats.effect.internals.IORunLoop$.start(IORunLoop.scala:34)
cats.effect.internals.IOBracket$$anonfun$apply$1.apply(IOBracket.scala:44)
cats.effect.internals.IOBracket$$anonfun$apply$1.apply(IOBracket.scala:34)
cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:337)
cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:351)
cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:372)
cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:312)
cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
java.lang.Thread.run(Thread.java:745)
```

**Comment by indig** [comment]
Spoke with @jeremyrsmith and this might be related to Spark running 2.11.8 while we use 2.11.11

**Comment by jeremyrsmith** [comment]
Actually, I took a look at the assembly JAR and it doesn't include the Scala library. Could the dependencies in that notebook have been bringing in the Scala library (of a different version) inside of an uber-JAR?

**Comment by jeremyrsmith** [comment]
Fixed by #133
